dev_guide/deployments.adoc

```diff
@@ -273,6 +273,14 @@ image stream changes and the new tag value differs from the current image
 specified in the deployment configuration's `helloworld` container, a new
 deployment is created using the new tag value for the `helloworld` container.
 
+The following table explains the possible combinations between ConfigChange
+and ImageChange triggers.
+
+  Trigger                  | true                                      | false
+  -----------------------  | ----------------------------------------- | ------------------------------------------------------
+  ConfigChange             | Deployments are started automatically     | Deployments need to be started manually
+  ImageChange (automatic)  | Images are always resolved automatically  | Only the initial image will be resolved in the template
```

Comment:
Bruno, can you retry what you are doing on HEAD of Origin and report back
what you observe?
The two controllers in play have been refactored since your working version.

> so openshift doesn't "know" about them

Honestly, I don't understand this part. Your image should be in an image
stream so OpenShift
knows about it, right?

On Tue, May 31, 2016 at 4:51 PM, Bruno Barcarol Guimarães <
notifications@github.com> wrote:

> In dev_guide/deployments.adoc
> <https://github.com/openshift/openshift-docs/pull/2150#discussion_r65197265>
> :
>
> > @@ -273,6 +273,14 @@ image stream changes and the new tag value differs from the current image
> >  specified in the deployment configuration's `helloworld` container, a new
> >  deployment is created using the new tag value for the `helloworld` container.
> >
> > +The following table explains the possible combinations between ConfigChange
> > +and ImageChange triggers.
> > +
> > +  Trigger                  | true                                      | false
> > +  -----------------------  | ----------------------------------------- | ------------------------------------------------------
> > +  ConfigChange             | Deployments are started automatically     | Deployments need to be started manually
> > +  ImageChange (automatic)  | Images are always resolved automatically  | Only the initial image will be resolved in the template
>
> @rhcarvalho <https://github.com/rhcarvalho>: (sorry, I was out last week)
> my issue is the opposite: the images
> are already on the node (they aren't built, so openshift doesn't "know"
> about
> them) and the only thing I need is for the deployment to start when the DC
> is
> created. If I only have the ConfigChange trigger, that's what happens. But
> when I add an ImageChangeTrigger, I get this on the log:
>
> Couldn't create initial deployment for deploymentConfig "x/y": DeploymentConfig "y" is invalid: triggers[1].imageChange.tag: Invalid value: "latest": no image recorded for x/y:latest
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/2150/files/97c25356b03bef326537fbf7aa454100ec58a53a#r65197265>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADuFfzkD_vpCSRejQeOH30Q-eeJuWVTrks5qHEsCgaJpZM4ImsqM>
> .
>